File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.ts
class RandomizedSet {
    values: number[];
    index: Map<number, number>;

    constructor() {
        this.values = [];
        this.index = new Map();
    }

    insert(val: number): boolean {
        if (this.index.has(val)) return false;
        this.index.set(val, this.values.length);
        this.values.push(val);
        return true;
    }

    remove(val: number): boolean {
        if (!this.index.has(val)) return false;
        const position = this.index.get(val);
        const last = this.values[this.values.length - 1];
        this.values[position] = last;
        this.index.set(last, position);
        this.values.pop();
        this.index.delete(val);
        return true;
    }

    getRandom(): number {
        const i = Math.floor(Math.random() * this.values.length);
        return this.values[i];
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * var obj = new RandomizedSet()
 * var param_1 = obj.insert(val)
 * var param_2 = obj.remove(val)
 * var param_3 = obj.getRandom()
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.py
import random

class RandomizedSet:

    def __init__(self):
        self.values = []
        self.index = {}

    def insert(self, val: int) -> bool:
        if val in self.index:
            return False
        self.index[val] = len(self.values)
        self.values.append(val)
        return True
        
    def remove(self, val: int) -> bool:
        if val not in self.index:
            return False
        position = self.index[val]
        last = self.values[-1]
        self.values[position] = last
        self.index[last] = position
        self.values.pop()
        del self.index[val]
        return True

    def getRandom(self) -> int:
        return random.choice(self.values)
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()